src/factories/beacon/InverterBeacon.sol

```diff
@@ -0,0 +1,150 @@
+
+// External Interfaces
+import {IBeacon} from "@oz/proxy/beacon/IBeacon.sol";
+
+//Internal Interfaces
+import {IInverterBeacon} from "src/factories/beacon/IInverterBeacon.sol";
+/**
+ * @title Beacon
+ */
+
+contract InverterBeacon is IInverterBeacon, ERC165, Ownable2Step {
+    //--------------------------------------------------------------------------------
+    // ERC-165 Public View Functions
+
+    /// @inheritdoc ERC165
+    function supportsInterface(bytes4 interfaceId)
+        public
+        view
+        virtual
+        override
+        returns (bool)
+    {
+        return interfaceId == type(IInverterBeacon).interfaceId
+            || interfaceId == type(IBeacon).interfaceId
+            || super.supportsInterface(interfaceId);
+    }
+
+    //--------------------------------------------------------------------------------
+    // State
+
+    /// @dev The beacon's implementation address.
+    address private _implementation;
+
+    /// @dev The beacon's current implementation pointer.
+    address private _currentImplementation;
+
+    /// @dev Is the beacon shut down / in emergency mode
+    bool public _emergencyMode;
+
+    /// @dev The major version of the implementation
+    uint private majorVersion;
+
+    /// @dev The minor version of the implementation
+    uint private minorVersion;
+
+    //--------------------------------------------------------------------------
+    // Constructor
+
+    constructor(uint _majorVersion) Ownable(_msgSender()) {
+        majorVersion = _majorVersion;
+    }
+
+    //--------------------------------------------------------------------------------
+    // Public View Functions
+
+    /// @inheritdoc IBeacon
+    function implementation() public view virtual override returns (address) {
+        return _currentImplementation;
+    }
+
+    /// @inheritdoc IInverterBeacon
+    function emergencyModeActive() external view returns (bool) {
+        return _emergencyMode;
+    }
+
+    /// @inheritdoc IInverterBeacon
+    function version() external view returns (uint, uint) {
+        return (majorVersion, minorVersion);
+    }
+
+    //--------------------------------------------------------------------------------
+    // onlyOwner Mutating Functions
+
+    /// @inheritdoc IInverterBeacon
+    function upgradeTo(
+        address newImplementation,
+        uint newMinorVersion,
+        bool overrideShutdown
+    ) public onlyOwner {
+        if (newMinorVersion <= minorVersion) {
+            revert Beacon__InvalidImplementationMinorVersion();
+        }
+
+        _setImplementation(newImplementation, overrideShutdown);
+
+        minorVersion = newMinorVersion;
+
+        emit Upgraded(newImplementation, newMinorVersion);
+    }
+
+    //--------------------------------------------------------------------------------
+    // onlyOwner Intervention Mechanism
+
+    /// @inheritdoc IInverterBeacon
+    function shutdownImplementation() external {
```

Comment:
Pedantic-Mode still active: As you "shut down" the implementation and not "shutdown"; shouldn't this be camel-cased as `shutDownImplementation` instead of `shutdownImplementation`?

Reply:
I think you can write it both ways actually 
I had it in camelcase before and it looked kinda weird to me
I would keep it the way it was before :P

Reply:
Lol I looked it up again and shutdown as a verb is really not a thing :see_no_evil: 
Fuuu
Will change it then